### PR TITLE
refactor: fix + refactor JSON file loading within new caching API

### PR DESCRIPTION
**Pull request by cameronmeissner**
**What type of PR is this?**

**What this PR does / why we need it**:

currently we try to load in the contents of both manifest/components JSON files as part of initialization to support the new VHD cache api. this doesn't seem to be working with the current implementation, thus this PR migrates this logic towards using the new `parts` package to read these files instead, introduced in this PR: https://github.com/Azure/AgentBaker/pull/4215. 

to accomplish this, we need to include `components.json` within the emedded FS rooted at `parts/` - as a result this PR moves `components.json` to `parts/linux/cloud-init/artifacts/components.json`, and adds a new symlink to the same within `vhdbuilder/packer`

also includes some light refactoring of the caching API itself

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/agent/vhd/cache/cache.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+const (
+	manifestFilePartPath   = "linux/cloud-init/artifacts/manifest.json"
+	componentsFilePartPath = "linux/cloud-init/artifacts/components.json"
+)
+
+//nolint:gochecknoinits
+func init() {
+	if err := initVHDCacheContent(); err != nil {
+		panic(err)
```
Comment (Devinwong) [inline pkg/agent/vhd/cache/cache.go:23]: Should we use panic or just returning error here? Does the panic occur during VHD baking? It won't crash during node provisioning, right?

Comment (cameronmeissner) [reply pkg/agent/vhd/cache/cache.go:23]: we panic since this is in an `init()` - https://tutorialedge.net/golang/the-go-init-function/

Comment (Devinwong) [reply pkg/agent/vhd/cache/cache.go:23]: Just to clarify a bit, who is importing/using this cache package?

Comment (cameronmeissner) [reply pkg/agent/vhd/cache/cache.go:23]: the `agent` pkg is the primary consumer, which contains the bakerapi code

**Review by Devinwong — APPROVED** [review]
lgtm

**Comment by cameronmeissner** [comment]
> lgtm since this mostly just moves the logic around, I wonder if there was a better way to actual create a symlink though, non blocking!

I think in a subsequent PR we can just completely remove the symlink altogether and just move components.json over to parts/ - that will entail changing a bunch of other references and I wanted to keep this PR on the smaller side if possible

**Review by lilypan26 — APPROVED** [review]

File: pkg/agent/vhd/cache/cache.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	manifestContent = bytes.ReplaceAll(manifestContent, []byte("#EOF"), []byte(""))
+	var manifest Manifest
+	if err = json.Unmarshal(manifestContent, &manifest); err != nil {
+		return nil, fmt.Errorf("unmarshalling manifest.json file part content: %w", err)
+	}
+	return &manifest, nil
+}
+
+func getComponents() (*Components, error) {
+	componentsContent, err := parts.Templates.ReadFile(componentsFilePartPath)
```
Comment (cameronmeissner) [inline pkg/agent/vhd/cache/cache.go:100]: example

**Pull request merged**